### DISCUSSION TITLE
static file: Symlinks that point to outside of static directory

### DISCUSSION THREAD

**Discussion by skirino** [encode/starlette#1594] [Potential Issue] [answered]
Hi,

The current version (we are using 0.19.0) of starlette staticfile resolves symlink's path and checks if the resolved path resides in the static file directory.
relevant code: `lookup_path()` https://github.com/encode/starlette/blob/master/starlette/staticfiles.py#L156-L170
This rejects valid symlinks that reside in the static directory but point to files outside of static directory. Is it OK to quit using `realpath()` and instead use e.g. `normpath()` (string manipulation without syscall)? If yes I'll be happy to submit a PR.

cf. flask's implementation does not resolve symlink's realpath.
- https://github.com/pallets/werkzeug/blob/main/src/werkzeug/utils.py#L547-L589
- https://github.com/pallets/werkzeug/blob/main/src/werkzeug/security.py#L110-L140

Our context:
We are uisng [bazel](https://bazel.build) and [rules_python](https://github.com/bazelbuild/rules_python), and we have [dagster](https://dagster.io) project. When running dagit (dagster's web UI) using bazel, bazel prepares symlinks for each of dagit's source files (including .py, .js, etc.). The symlinks for .js and .css happen to live outside of the static directory and cannot be returned to web browser (404 error).

**Comment by aminalaee** [comment]
Hey,
If I get your point correctly, there's already an open PR for this: https://github.com/encode/starlette/pull/1377
Feel free to review it.

**Reply by skirino** [reply]
Thank you for the response and the PR! Sorry I failed to search the issue/PR.

**Comment by Kludex** [comment, marked as answer]
This was solved at some point.
